hdfs/src/main/scala/akka/stream/alpakka/hdfs/scaladsl/HdfsSource.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.hdfs.scaladsl
+
+import akka.NotUsed
+import akka.stream.IOResult
+import akka.stream.scaladsl.{Source, StreamConverters}
+import akka.util.ByteString
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.hadoop.io.compress.CompressionCodec
+import org.apache.hadoop.io.{SequenceFile, Writable}
+
+import scala.concurrent.Future
+
+object HdfsSource {
+
+  /**
+   * Scala API: creates a [[Source]] that consumes as [[ByteString]]
+   *
+   * @param fs Hadoop file system
+   * @param path the file to open
+   * @param chunkSize the size of each read operation, defaults to 8192
+   */
+  def data(
+      fs: FileSystem,
+      path: Path,
+      chunkSize: Int = 8192
+  ): Source[ByteString, Future[IOResult]] =
+    StreamConverters.fromInputStream(() => fs.open(path), chunkSize)
+
+  /**
+   * Scala API: creates a [[Source]] that consumes as [[ByteString]]
+   *
+   * @param fs Hadoop file system
+   * @param path the file to open
+   * @param codec a streaming compression/decompression pair
+   * @param chunkSize the size of each read operation, defaults to 8192
+   */
+  def compressed(
+      fs: FileSystem,
+      path: Path,
+      codec: CompressionCodec,
+      chunkSize: Int = 8192
+  ): Source[ByteString, Future[IOResult]] =
+    StreamConverters.fromInputStream(() => codec.createInputStream(fs.open(path)), chunkSize)
```

Comment:
A source created by `fromInputStream` already runs on the IO dispatcher, therefore it is fine here.